cmd/podman/artifact/push.go

```diff
@@ -0,0 +1,234 @@
+package artifact
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/containers/buildah/pkg/cli"
+	"github.com/containers/common/pkg/auth"
+	"github.com/containers/common/pkg/completion"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/podman/v5/cmd/podman/common"
+	"github.com/containers/podman/v5/cmd/podman/registry"
+	"github.com/containers/podman/v5/pkg/domain/entities"
+	"github.com/containers/podman/v5/pkg/util"
+	"github.com/spf13/cobra"
+)
+
+// pushOptionsWrapper wraps entities.ImagepushOptions and prevents leaking
+// CLI-only fields into the API types.
+type pushOptionsWrapper struct {
+	entities.ArtifactPushOptions
+	TLSVerifyCLI               bool // CLI only
+	CredentialsCLI             string
+	SignPassphraseFileCLI      string
+	SignBySigstoreParamFileCLI string
+	EncryptionKeys             []string
+	EncryptLayers              []int
+	DigestFile                 string
+}
+
+var (
+	pushOptions     = pushOptionsWrapper{}
+	pushDescription = `Push an OCI artifact from local storage to an image registry`
+
+	pushCmd = &cobra.Command{
+		Use:   "push [options] ARTIFACT.",
+		Short: "Push an OCI artifact",
+		Long:  pushDescription,
+		RunE:  artifactPush,
+		// PersistentPreRunE: devOnly,
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: common.AutocompleteArtifacts,
+		Example:           `podman artifact push quay.io/myimage/myartifact:latest`,
+		// TODO Autocomplete function needs to be done
+	}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: pushCmd,
+		Parent:  artifactCmd,
+	})
+	pushFlags(pushCmd)
+
+	// TODO When the inspect structure has been defined, we need to uncommand and redirect this.  Reminder, this
+	// will also need to be reflected in the podman-artifact-inspect man page
+	// _ = inspectCmd.RegisterFlagCompletionFunc(formatFlagName, common.AutocompleteFormat(&machine.InspectInfo{}))
```

Comment:
I know this is a comment, but it might be a bug that will trip later:
```suggestion
	// _ = inspectCmd.RegisterFlagCompletionFunc(formatFlagName, common.AutocompleteFormat(&artifact.InspectInfo{}))
```